### DISCUSSION TITLE
QR code & barcode reader as an option on the keyboard

### DISCUSSION THREAD

**Discussion by Tsjoecha** [AnySoftKeyboard/AnySoftKeyboard#3559] [Ideas]
Hi,

I found on the great world wide web a review which indicates AnySoftKeyboard is the best opensource alternative for the well known keyboards as gboard, swiftkey,... 
Those keyboards lack a feature which would be very useful for me:  I'm looking for a good combination of a textual keyboard with the option to read qr codes and barcodes in general as well. I know this feature isn't available here as well, but I think it would be a huge plus if it would. Is there any chance to implement this?
Thanks in advance!

**Comment by menny** [comment]
Yes, this has been suggested a few times, and it is something I want to add.

Unfortunately, this is not a small thing to implement (mostly, to create the plumbing in the keyboard to support this flow) and I lack the time to sit down and do it.
